base/combinatorics.jl

```diff
@@ -2,38 +2,40 @@
 
 # Factorials
 
-const _fact_table64 = Vector{Int64}(undef, 20)
-_fact_table64[1] = 1
-for n in 2:20
-    _fact_table64[n] = _fact_table64[n-1] * n
-end
+const _shifted_fact_table64 = [1; cumprod(1:Int64(20))]  # tables shifted by 1 so factorial(0) can be looked up; maybe OffsetArray better?
 
-const _fact_table128 = Vector{UInt128}(undef, 34)
-_fact_table128[1] = 1
-for n in 2:34
-    _fact_table128[n] = _fact_table128[n-1] * n
-end
+const _shifted_fact_table128 = [1; cumprod(1:UInt128(34))]
 
-function factorial_lookup(n::Integer, table, lim)
+const _fact_table64 = [cumprod(1:Int64(20))]  # kept just for gamma function (Base._fact_table64) in SpecialFunctions.jl, notused here
+
```

Comment:
We can update the `gamma` function in SpecialFunctions.jl, and not keep this extra definition. Let's focus on making sure this PR does give a speed bump, and if it does, we can change `gamma`.

Reply:
I tried to update gamma, but even with it changed, CI complains so this needs to be in.